tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -0,0 +1,313 @@
+
+    @pytest.fixture
+    def dataframe(self):
+        spark = SparkSession.builder.getOrCreate()  # pyright: ignore
+
+        data = [('36636', 'Finance', (3000, 'USA')), ('40288', 'Finance', (5000, 'IND')),
+                ('42114', 'Sales', (3900, 'USA')), ('39192', 'Marketing', (2500, 'CAN')),
+                ('34534', 'Sales', (6500, 'USA'))]
+        schema = StructType([
+            StructField('id', StringType(), True),
+            StructField('dept', StringType(), True),
+            StructField(
+                'properties',
+                StructType([
+                    StructField('salary', IntegerType(), True),
+                    StructField('location', StringType(), True)
+                ]))
+        ])
+
+        df = spark.createDataFrame(data=data, schema=schema).repartition(3)
+        yield df
+
+    @pytest.mark.parametrize('keep_local', [True, False])
+    @pytest.mark.parametrize('merge_index', [True, False])
+    def test_end_to_end_conversion_local_nocolumns(self, dataframe: Any, keep_local: bool,
+                                                   merge_index: bool):
+        out = mkdtemp()
+        mds_kwargs = {
+            'out': out,
+            'keep_local': keep_local,
+        }
+
+        with pytest.raises(ValueError):  # , match=f'* is not supported by MDSwriter'):
+            _, _ = dataframeToMDS(dataframe.select(col('id'), col('dept'), col('properties')),
+                                  merge_index=merge_index,
+                                  mds_kwargs=mds_kwargs)
+
+        _, _ = dataframeToMDS(dataframe.select(col('id'), col('dept')),
+                              merge_index=merge_index,
+                              mds_kwargs=mds_kwargs)
+
+        assert (len(os.listdir(out)) > 0), f'{out} is empty'
+        for d in os.listdir(out):
+            if os.path.isdir(os.path.join(out, d)):
+                assert (os.path.exists(os.path.join(
+                    out, d, 'index.json'))), f'No index.json found in subdirectory {d}'
+
+        if merge_index:
+            assert (os.path.exists(os.path.join(out, 'index.json'))), 'No merged index.json found'
+            mgi = json.load(open(os.path.join(out, 'index.json'), 'r'))
+            nsamples = 0
+            for d in os.listdir(out):
+                sub_dir = os.path.join(out, d)
+                if os.path.isdir(sub_dir):
+                    shards = json.load(open(os.path.join(sub_dir, 'index.json'), 'r'))['shards']
+                    if shards:
+                        nsamples += shards[0]['samples']
+            assert (nsamples == sum([a['samples'] for a in mgi['shards']]))
+        else:
+            assert not (os.path.exists(os.path.join(
+                out, 'index.json'))), 'merged index is created when merge_index=False'
+
+    @pytest.mark.parametrize('use_columns', [True, False])
+    def test_end_to_end_conversion_local_decimal(self, decimal_dataframe: Any, use_columns: bool):
+        out = mkdtemp()
```

Comment:
Same here. Please try to use [this](https://github.com/mosaicml/streaming/blob/main/tests/common/utils.py#L19) method and all the other places in this file.

Reply:
Please see reply above.